### PR TITLE
ASM-6515 ASM deployment to support EMC Storage provisioning for HyperV FC

### PR DESCRIPTION
**Pull request by sushilrai**
Additional facts required for storage groups and initiators required for EMC asm-deployer changes

### CODE REVIEW

File: lib/puppet/provider/vnx_storagegroup/vnx_storagegroup.rb [hunk 1]
```diff
@@ -206,7 +210,9 @@ def set_storagegroup
                         []
                       else
                         pairs.map{|pair| pair.values_at('hlu', 'alu').map &:to_s}.sort
-                      end
+                     end
+      require 'pry'
+      binding.pry
```
Comment (ripienaar) [inline lib/puppet/provider/vnx_storagegroup/vnx_storagegroup.rb]: this shouldnt be here I think?

File: lib/puppet/provider/vnx_storagegroup/vnx_storagegroup.rb [hunk 2]
```diff
@@ -26,8 +26,12 @@ def addhlu hlu, alu
...
     #   run ["lun", "-create", "-l", alu]
     # end
-    args = ["storagegroup", "-addhlu", "-gname", resource[:sg_name], "-hlu", hlu, "-alu", alu]
-#    puts "#{args}"
+    if hlu.empty?
+      args = ["storagegroup", "-addhlu", "-gname", resource[:sg_name], "-alu", alu]
+    else
+      args = ["storagegroup", "-addhlu", "-gname", resource[:sg_name], "-hlu", hlu, "-alu", alu]
+    end
+    Puppet.debug "#{args}"
```
Comment (ripienaar) [inline lib/puppet/provider/vnx_storagegroup/vnx_storagegroup.rb]: this debug message is not going to help anyone - a array stuck in the middle of nowhere with no context?

File: lib/puppet/provider/vnx_initiator/vnx_initiator.rb [hunk 3]
```diff
@@ -76,38 +76,47 @@ def get_instances
...
+            next
+          end
+        end
+      end
       if port["storage_group"]
         gname = port["storage_group"]
-      #else
-      #  gname = create_temp_storage_group
+        #else
+        #  gname = create_temp_storage_group
```
Comment (ripienaar) [inline lib/puppet/provider/vnx_initiator/vnx_initiator.rb]: please remove

**Comment by ripienaar** [comment]
hmm, i cant merge here either, you can merge it @sushilrai

**Comment by sushilrai** [comment]
@ripienaar I have added the permission for you. Please check

**Pull request merged**